### ISSUE TITLE
[BUG] cudf::ast Module Fails Implicit Cast from Decimal to Double Due to Recent Explicit Conversion Requirement (PR #15438)

### ISSUE THREAD

**Issue by aocsa** [rapidsai/cudf#16023]
**Describe the bug**

A recent pull request (PR) [#15438](https://github.com/rapidsai/cudf/pull/15438) introduced a breaking change requiring that Floating <--> Fixed-Point conversions must be called explicitly, thereby disallowing implicit casts from decimal to double. Consequently, the cudf::ast module, which still employs implicit casting, is affected.

While cuDF version 24.06 has incorporated these changes into various cuDF functions, the necessary updates to the cudf::ast module were missed. This oversight results in the failure of implicit casts from decimal to double within the cudf::ast module, as highlighted in the following code: 

https://github.com/rapidsai/cudf/blob/3cb3df3255efaec4a5ebb6cb7606067f753e3554/cpp/include/cudf/ast/detail/operators.hpp#L825

**Steps/Code to reproduce bug**

```
TEST_F(TransformTest, BasicAdditionDoubleCast)
{
  auto c_0   = column_wrapper<double>{3, 20, 1, 50};
  //auto c_1   = column_wrapper<int8_t>{10, 7, 20, 0};
  std::vector<__int128_t> data1{10, 7, 20, 0};  
  auto c_1 = cudf::test::fixed_point_column_wrapper<__int128_t>(
                            data1.begin(), data1.end(), numeric::scale_type{0});
  auto table = cudf::table_view{{c_0, c_1}};
  auto col_ref_0  = cudf::ast::column_reference(0);
  auto col_ref_1  = cudf::ast::column_reference(1);
  auto cast       = cudf::ast::operation(cudf::ast::ast_operator::CAST_TO_FLOAT64, col_ref_1);
  auto expression = cudf::ast::operation(cudf::ast::ast_operator::ADD, col_ref_0, cast);
  auto expected = column_wrapper<double>{13, 27, 21, 50};
  auto result   = cudf::compute_column(table, expression);
  CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected, result->view(), verbosity);
}
```

This unit test results in an exception:

`C++ exception with description "CUDF failure at:/opt/mambaforge/conda-bld/libcudf-ext_1717805640036/work/cpp/include/cudf/ast/detail/operators.hpp:1063: Invalid unary operation." thrown in the test body.`

**Expected behavior**

This test is expected to pass as it functions correctly in cuDF version 24.04.

Environment: conda 
Method of cuDF install: source code
v24.06.00 branch release

**Comment by bdice** [comment]
@aocsa Thank you for the issue report!

@pmattione-nvidia Would you be able to take a look at this?

**Comment by bdice** [comment]
I realized that we missed adding a test case for this in #16038. cc: @pmattione-nvidia

**Comment by cryos** [comment]
Could this fix be considered for inclusion in a 24.06 hotfix release?

**Comment by vyasr** [comment]
Yup, we are currently discussing what will be in our hotfix and the fixes for this issue are on the list.
